File: primatis-api/src/main/java/be/primatis/user/web/ResidenceResponse.java
package be.primatis.user.web;

import be.primatis.user.Residence;

import java.time.LocalDate;

/**
 * Contrat REST de lecture d'une {@code Residence} (DEV-05.8, résidence
 * courante ou entrée d'historique). {@code endDate} est {@code null} pour la
 * résidence courante. N'expose jamais {@code UserResponse} (DEV-05.8-DEC-10 :
 * le lien inverse n'est pas nécessaire, la ressource est déjà accédée depuis
 * un utilisateur identifié par l'URL).
 */
public record ResidenceResponse(Long id, AddressResponse address, LocalDate startDate, LocalDate endDate) {

    public static ResidenceResponse from(Residence residence) {
        return new ResidenceResponse(
                residence.getId(),
                AddressResponse.from(residence.getAddress()),
                residence.getStartDate(),
                residence.getEndDate());
    }
}

File: primatis-web/src/app/user/models/residence-response.ts
import { AddressResponse } from './address-response';

/**
 * Voir `be.primatis.user.web.ResidenceResponse` côté backend. `endDate`
 * `null` désigne la résidence courante. Ressource distincte de
 * `UserResponse`/`MeProfileResponse` — jamais imbriquée dans l'un ou
 * l'autre.
 */
export interface ResidenceResponse {
  id: number;
  address: AddressResponse;
  startDate: string;
  endDate: string | null;
}
